x86_64/src/arch/devcons.rs

```diff
@@ -0,0 +1,42 @@
+// Racy to start.
+
+use core::fmt;
+use port::devcons::ConsoleDriver;
+
+struct Console {
+    port: u16,
+}
+
+impl ConsoleDriver for Console {
+    fn uartputb(&self, b: u8) {
+        crate::x86_64::uart16550::putb(self.port, b);
+    }
+}
+
+// It would be nice if most the below code was in port....
+
+impl fmt::Write for Console {
```

Comment:
See the comments on the port code. There's no reason Not to move Console, the impl for `fmt::Write` and print/println to port.

Reply:
The reason Console and the print() function weren't moved into port is that they have architecture-specific details (the uart port).

With a bit of refactoring, most of the functions can be moved to port I think, but the Console object needs to be created somewhere other than in the print() function.

Reply:
Right. If Console is implemented in terms of a portable trait object, instead, then that dependency is severed.